main.cpp

```diff
@@ -1 +1,181 @@
+/*
+ * Baquerizo Lema Anny Emily
+ * Rendu le 29 septembre 2022
+ * Laboratoire Tic Tac Toe
+ *
+ */
 #include <iostream>
+#include <string>
+#include <list>
+
+using namespace std;
+
+bool check(int valeur, list<int> liste, bool& a)    {
+    for (int i : liste) {
+        if(valeur == i)  {
+            a = true;
+        }
+    }
+    return a;
+}
+
+void plateau(string (&qua)[3][3]){
+    cout << "   " + qua[0][0] + "   |   " + qua[0][1] + "   |   " + qua[0][2] + ""<< endl;
+    cout << "--------------------" << endl;
+    cout << "   " + qua[1][0] + "   |   " + qua[1][1] + "   |   " + qua[1][2] + "" << endl;
+    cout << "--------------------" << endl;
+    cout << "   " + qua[2][0] + "   |   " + qua[2][1] + "   |   " + qua[2][2] + ""<< endl;
+    cout << "" << endl;
+}
+
+//Version pour 2 joueurs V2.2: changement de joueur effectué via un bool
+void player2 () {
+    string qua[3][3] = {
+            {"", "", ""},
+            {"", "", ""},
+            {"", "", ""}
+    };
+
+    bool end = false;
+    //Fonctionnement
+    cout << "Pour choisir une case, il suffit de rentrer le chiffre de la case correspondante lors de votre tour." << endl;
+    cout << "" << endl;
+    cout << "   1   |   2   |   3   "<< endl;
+    cout << "-----------------------" << endl;
+    cout << "   4   |   5   |   6   " << endl;
+    cout << "-----------------------" << endl;
+    cout << "   7   |   8   |   9   "<< endl;
+    cout << "" << endl;
+
+
+    //Initialisation nmbre de tour
+    int tour = 0;
+
+    //Changement de tour du joueur
+    bool joueur1 = true;
+
+    //Liste de case restante
+    list<int> libre = {1, 2, 3, 4, 5, 6, 7, 8, 9};
+
+    cout << "La partie commence" << endl;
+    //1 Tour de jeu
+    while (!end && tour <= 8) {
+        //Tour
+        tour ++;
+        cout << "Tour " << tour << endl;
+
+        //Affichage plateau de jeu
+        plateau(qua);
+
+        string coup;
+
+        //Tour du joueur
+        string symbole;
+
+        if (joueur1)   {
+            cout << "Au tour du joueur 1 :   " << endl;
+            symbole = "o";
+        }
+        else    {
+            cout << "Au tour du joueur 2 : " << endl;
+            symbole = "x";
+        }
+        cin >> coup;
+        int choix = stoi(coup);
+
+        bool vide = false;
+        check(choix, libre, vide);
```

Comment:
Passage par référence rend ceci très illisible. Mieux:
```suggestion
        bool vide = check(choix, libre);
```